### ISSUE TITLE
Is there any way to find the largest horizon rectangle?

### ISSUE THREAD

**Issue by kaka1909** [OpenStitching/lir#27]
First of all, many thanks for your project which help me a lot in my work. And I have a question that, is there any way to find the largest horizon rectangle inside a shape? I need a horizon rectangle to put some words in it. 

Thanks,
Mi

**Comment by lukasalexanderweber** [comment]
Can you make a sketch to describe what excatly you need ?

**Comment by kaka1909** [comment]
> Can you make a sketch to describe what excatly you need ?

Yes. Actually I am trying to find a large horizontal "empty" area to put some words on it. Like below:

![mihai_rect_副本](https://github.com/OpenStitching/lir/assets/6416440/16e29520-4888-49e9-b7f8-7106e6f0c339)


The blue rectangle is the largest rectangle computed by lir, but if I put some words in it, it is too narrow. So maybe the green rectangle which is indeed smaller than blue one is better for my needs.

**Comment by lukasalexanderweber** [comment]
I think there should be the possibility to return multiple rectangles ordered by size

**Comment by kaka1909** [comment]
Yes, that will be great.

**Comment by warriorstar-orion** [comment]
I would love this as well. As the reporter said, having the widest available rectangle is very useful for finding where text can be placed. If a list of rectangles is returned I imagine you don't even really need to sort them, the caller can sift through them to find the widest themselves.
